docs/development.md

```diff
@@ -55,3 +55,131 @@ Pre-commit should now work as expected when making commits even without the need
 an active poetry shell. You can also manually run pre-commit (e.g. `pre-commit run -a`)
 and the unit tests with `pytest`. Remember you'll need to prefix these with `poetry run`
 first if you don't have an active poetry shell.
+
+## Docker Development Stack
+
+The Docker Compose stack is defined in `docker-compose.yml` and provides the services
+required to enable development of the web application. The services are described in detail below.
+
+### kafka
+
+Kafka is a "distributed event streaming platform" used as a channel to pass messages
+between different components of the DUNE control system. It is primarily used to send
+informational messages for consumption by system operators. A key role of the UIs we are
+developing is to present these messages to system operators to help them identify and
+troubleshoot issues.
+
+<!-- markdownlint-disable next-line code-block-style -->
+!!! note
+
+    Messages sent via Kafka are referred to as "broadcast messages" in the drunc
+    codebase and documentation.
+
+Kafka is known for being complex to set up and manage. The Docker Compose stack provides
+a Kafka image packaged by Bitnami which seems to have worked reliably so far.
+Configuration for the Kafka service is carried out using environment variables in
+`docker-compose.yml`. The configuration values used are not well understood so it is not
+recommended to change them unless systemic issues are encountered.
+
+Longer term the DUNE project may move away from Kafka to a more lightweight solution.
+The web application should aim to be as agnostic as possible to the underlying messaging
+system.
+
+### drunc
+
+This service is only used as part of the `drunc` Docker Compose profile. See the [Docker
+Setup overview] for details.
+
+Runs the various components of the DUNE control system that the web app interfaces with.
+Drunc has a complex set of dependencies collectively referred to the as the DUNE DAQ
+environment (see the [DUNE DAQ documentation] for more information). The upshot is that
+it is not feasible for us to build our own Drunc image from scratch for development.
+Instead we use a base image published by the Dune project that provides the required
+dependencies.
+
+The base images are published by the [daq-release repository]. There are separate images
+for nightly ([nightly-release-alma9]) and release ([frozen-release-alma9]) builds. We
+are currently tracking nightly builds as we are dependent on cutting edge drunc
+developments. It may be feasible in the future to swap to a pinned release build.
+
+To ensure a consistent developer experience we pin the development image to a fixed base
+image. The base image is manually tagged and pushed to the
+[imperialcollegelondon/dunedaq_dev_environment] package with a tag including the date of
+the nightly.
```

Comment:
Why cannot we directly point to the chosen nightly build in our dockerfile and instead have to download the base image, rename and push again?

Reply:
The nightly images are not retained by the dune project so we have to take our own copy and point to that.